### PR TITLE
feat(typescript): add typescript support

### PR DESCRIPTION
**Pull request by vijaypemmaraju**
Check the new `examples/` folder for usage.

The best part is that the usage should be the same as before in javascript, but in typescript, adding the generic gets us that all those juicy type definitions 😍

### CODE REVIEW

**Comment by vijaypemmaraju** [comment]
FYI there's still some projects using the github import, so we have to keep the build directory around for now 😒

File: example/index.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+const types = {
+  FOO: {
+    BAR: 'BAR',
+    BAZ: standardActions(),
+  }
+};
+
+const actionTypes = createActions<typeof types>(types) as typeof types;
+
+console.log(actionTypes);
```
Comment (corytheboyd) [inline example/index.ts]: Nice, love the idea of an examples folder!

**Comment by corytheboyd** [comment]
We discussed this offline already, but summarizing here as well. If this is all becoming TypeScript, we should just ditch Webpack and simplify the build process to just use the TS compiler. To be done outside the scope of this PR.

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "compilerOptions": {
+    "outDir": "./dist/",
+    "noImplicitAny": true,
+    "module": "CommonJS",
+    "target": "es5",
+    "jsx": "react",
```
Comment (corytheboyd) [inline tsconfig.json]: Is this needed? Might be confusing to someone looking into our TS configuration

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    "allowSyntheticDefaultImports": true,
+    "allowJs": true,
```
Comment (corytheboyd) [inline tsconfig.json]: Same here, is there any JS left?

**Pull request merged**